src/server/search/doc_index.h

```diff
@@ -52,10 +52,83 @@ struct SearchResult {
   std::optional<facade::ErrorReply> error;
 };
 
-template <typename T> using SearchField = std::pair<T /*identifier*/, T /*short name*/>;
+/* SearchField represents a field that can store combinations of identifiers and aliases in various
+   forms: [identifier and alias], [alias and new_alias], [new identifier and alias] (used for JSON
+   data) This class provides methods to retrieve the actual identifier and alias for a field,
+   handling different naming conventions and resolving names based on the schema. */
+class SearchField {
+ private:
+  static bool IsJsonPath(std::string_view name) {
+    if (name.size() < 2) {
+      return false;
+    }
+    return name.front() == '$' && (name[1] == '.' || name[1] == '[');
+  }
+
+ public:
+  SearchField() = default;
+
+  SearchField(StringOrView name, bool is_short_name)
+      : name_(std::move(name)), is_short_name_(is_short_name) {
+  }
+
+  SearchField(StringOrView name, bool is_short_name, StringOrView new_alias)
```

Comment:
what's a "new alias" ?

Reply:
```
JSON.SET j1 . '{"value":1}'
FT.CREATE index ON JSON SCHEMA $.value AS some_alias NUMERIC
FT.AGGREGATE index "*" LOAD 1 @some_alias AS another_new_alias
> 1) (integer) 1
> 2) 1) "another_new_alias"
>      2) (integer) 1
```

`another_new_alias` is new alias. Here we are storing field like [name = "some_alias", is_short_name=true, new_alias="another_new_alias"] and we need to find identifier "$.value" for it.